app/acquisition_adapters.py

```
@@ -0,0 +1,527 @@
+
+JsonTransport = Callable[[str, Mapping[str, str], float], tuple[bytes, Mapping[str, str]]]
+
+
+class AcquisitionError(RuntimeError):
+    pass
+
+
+def _default_json_transport(url: str, headers: Mapping[str, str], timeout: float) -> tuple[bytes, Mapping[str, str]]:
+    req = request.Request(url, headers=dict(headers), method="GET")
+    try:
+        with request.urlopen(req, timeout=timeout) as response:  # nosec - adapter URL is fixed by provider
```

Ruff (0.16.3)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)